src/native/managed/cdac/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/DebugInfo/DebugInfoHelpers.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using ILCompiler.Reflection.ReadyToRun;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts;
+
+/// <summary>
+/// Shared bounds decoding helpers for DebugInfo contract versions.
+/// V1: 2-bit enumerated source type.
+/// V2: 3-bit flags (CallInstruction, StackEmpty, Async).
+/// </summary>
+internal static class DebugInfoHelpers
+{
+    internal static IEnumerable<OffsetMapping> DoBounds(NativeReader nativeReader, uint ilOffsetBias, uint sourceTypeBitCount)
```

Comment:
Maybe instead of uint sourceTypeBits we could pass the version number down here and do the right thing with it. That would make it easy to adjust the source type bits details in a future PR.